jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -625,9 +634,74 @@ def _update_stretch_histogram(self, msg={}):
             # in case only the sample has changed but its length has not,
             # we'll force the traitlet to trigger a change
             hist_mark.send_state('sample')
-
         self._stretch_histogram_needs_update = False
 
+    @observe('stretch_vmin_value', 'stretch_vmax_value', 'layer_selected',
+             'stretch_function_value', 'stretch_hist_nbins',
+             'image_contrast_value', 'image_bias_value')
+    def _update_stretch_curve(self, msg=None):
+        # Import here to prevent circular import (and not at the top of the method so the import
+        # check is avoided, whenever possible).
+        from jdaviz.configs.imviz.plugins.viewers import ImvizImageView
+        from jdaviz.configs.cubeviz.plugins.viewers import CubevizImageView
+
+        if (
+            not isinstance(self.viewer.selected_obj, (ImvizImageView, CubevizImageView)) or
+            not hasattr(self, 'stretch_histogram')
+        ):
+            # don't update histogram if selected viewer is not an image viewer,
+            # or the stretch histogram hasn't been initialized:
+            return
+
+        mark_label_prefix = "stretch_curve: "
+        for layer in self.layer.selected_obj:
+            # clear old mark, if it exists:
+            mark_label = f'{mark_label_prefix}{layer.label}'
+            mark_exists = mark_label in self.stretch_histogram.marks

```

Comment:
What is the benefit of tracking per-layer curves?  Just to future-proof if we ever support showing the histogram for multiple layers?  If we don't need that now, this can simplify greatly by creating the mark in the init with the rest of the other marks, _always_ modify that existing entry, and not have to worry about re-ordering the marks.

Reply:
the whole histogram would need to be updated for that though... is it worth the overhead now if we're not sure how that will work yet?

Reply:
If it's already implemented, what's the harm in keeping it?

Reply:
I think we will want a more general solution that handles the histogram in the same way and there is no good way to actually test this implementation as-is and it adds quite a bit of extra code that we're not 100% sure if we'll ever use 🤷‍♂️